docs/contributing.rst

```diff
@@ -6,8 +6,128 @@
+
+- `Uxarray documentation <https://uxarray.readthedocs.io/en/latest/?badge=latest>`_
+  houses significant documentation such as
+  `Getting Started <https://uxarray.readthedocs.io/en/latest/quickstart.html>`_,
+  `Installation <https://uxarray.readthedocs.io/en/latest/installation.html>`_,
+  `Contributor's Guide <https://uxarray.readthedocs.io/en/latest/contributing.html>`_
+  (i.e. this document), `Usage Examples <https://uxarray.readthedocs.io/en/latest/examples.html>`_,
+  `Tutorials <https://uxarray.readthedocs.io/en/latest/tutorials.html#>`_, and
+  `API Reference <https://uxarray.readthedocs.io/en/latest/api.html>`_.
+
+- `UXarray draft API <https://github.com/UXARRAY/uxarray/blob/main/docs/user_api/uxarray_api.md>`_
+  tentatively shows the eventual list of UXarray functionalities. It is
+  open to community feedback and inputs!
+
+  - Please create a
+    GitHub `Discussion <https://github.com/UXARRAY/uxarray/discussions>`_ or
+    `Issue <https://github.com/UXARRAY/uxarray/issues>`_ if you feel there
+    should be change in any way in this document!
+
+2. Working on the UXarray GitHub Repository
+-------------------------------------------
+
+In this section, we provide details about how to work on the UXarray
+GitHub repository. This is because most contributions, such as adding a new
+function or making changes to an existing one, writing a usage example, or
+making some modifications to this Contributor's Guide requires operating
+directly on the GitHub repository.
+
+Contributing to a GitHub repository follows almost the same process by
+any open development Python project maintained on GitHub.
+However, it can still seem complex and somewhat varied from one project to
+another. As such, we will provide an overview of GitHub here and refer the
+reader to other more comprehensive resources for detailed information about
+it (such as GitHub's this
+`Getting Started with Github <https://docs.github.com/en/get-started>`_
+guide), but we will provide great details about how to configure a GitHub
+development environment for UXarray development and how to contribute.
+
+2.1. Getting started with GitHub and Git
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Contributing to UXarray requires using GitHub, as already mentioned, and
+Git. The latter, Git, is an open source, command line tool for collaborative
+software version control, while GitHub is an online, web-accessible service
+that greatly simplifies using the powerful, yet often complex, Git.
+
+.. note::
+    GitHub operates entirely within a web browser. You do not need to
+    install anything, but you will need to set up a free GitHub account. Git
+    is a command line tool that most likely will need to be installed on
+    your machine and run from a "terminal" window, AKA a "shell".
+
+Using, and even just configuring, Git and GitHub are often the most
+daunting aspects of contributing to a GitHub hosted project. Here are
```

Comment:
Should there be a hyperlink here to the steps, or will the steps be listed below this section (and/or added at a later time)?

Reply:
The next subsections are those steps. I added ":" at the end of the paragraph now to reflect this.